frontend/src/components/pages/events/EventDetails/SignUpVariants.tsx

```diff
@@ -0,0 +1,243 @@
+ */
+
+type Props = {
+  event: Event;
+  user?: User | null;
+  loading: boolean;
+  extraInformation?: string;
+  onExtraInformationChange: (info: string) => void;
+  eventId: string;
+};
+
+const SignUpVariants: React.FC<Props> = ({
+  event,
+  user,
+  loading,
+  extraInformation,
+  onExtraInformationChange,
+  eventId,
+}) => {
+  const [snackbar, setSnackbar] = useState<
+    "SignUp" | "SignOff" | "OnWaitList" | "OffWaitList" | "SignUpError" | "SignOffError" | undefined
+  >(undefined);
+  const classes = useStyles();
+  const router = useRouter();
+
+  const { data: timeData, error: timeError } = useQuery(GET_SERVER_TIME, { fetchPolicy: "network-only" });
+  const { data: eventData, refetch: refetchEventData } = useQuery<{
+    event: Event;
+  }>(GET_EVENT, { variables: { id: eventId }, errorPolicy: "all" });
+  const [eventSignUp, { loading: signUpLoading, error: signUpError }] = useMutation<{
+    eventSignUp: { isFull: boolean };
+  }>(EVENT_SIGN_UP, {
+    onCompleted: () =>
+      refetchEventData({ eventId }).then((res) => {
+        res.data.event.attendable?.userAttendance?.isAttending ? setSnackbar("SignUp") : setSnackbar("OnWaitList");
+      }),
+    onError: () => setSnackbar("SignUpError"),
+  });
+  const [eventSignOff, { loading: signOffLoading }] = useMutation<{
+    eventSignOff: { isFull: boolean };
+  }>(EVENT_SIGN_OFF, {
+    onCompleted: () => {
+      refetchEventData({ eventId });
+      if (!eventData) return;
+      if (eventData.event.attendable?.userAttendance?.isAttending) setSnackbar("SignOff");
+      if (eventData.event.attendable?.userAttendance?.isOnWaitingList) setSnackbar("OffWaitList");
+    },
+    onError: () => setSnackbar("SignOffError"),
+  });
+
+  const handleClick = () => {
+    if (!user) return;
+    const userAttendance = eventData?.event.attendable?.userAttendance;
+    if (userAttendance?.isAttending || userAttendance?.isOnWaitingList) {
+      eventSignOff({
+        variables: { eventId },
+      });
+      return;
+    }
+    eventSignUp({
+      variables: { eventId, data: extraInformation },
+    });
+  };
+
+  const noPhoneNumberNorAlreadySignedUp =
+    !user?.phoneNumber &&
+    !event.attendable?.userAttendance?.isAttending &&
+    !event.attendable?.userAttendance?.isOnWaitingList;
+  const requiresExtraInfoAndNotAlreadySignedUp =
+    event.attendable?.hasExtraInformation &&
+    !event.attendable?.userAttendance?.isAttending &&
+    !event.attendable?.userAttendance?.isOnWaitingList;
+  const bindingSignupAndAlreadySignpedUp =
+    event.attendable?.bindingSignup && event.attendable?.userAttendance?.isAttending;
+  const requiresExtraInfoAndExtraInfoNotFilledIn =
+    event.attendable?.hasExtraInformation &&
+    !extraInformation &&
+    !event.attendable?.userAttendance?.isAttending &&
+    !event.attendable?.userAttendance?.isOnWaitingList;
+
+  if (!event.attendable) return null;
+
+  if (timeError || !timeData)
```

Comment:
`timeData` is expectedly `undefined` during load, which would cause this to be displayed during load.
```suggestion
  if (timeError)
```